kuma/api/signal_handlers.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+
+from kuma.wiki.models import Document
+from kuma.wiki.signals import render_done, restore_done
+
+from .tasks import publish, unpublish
+
+
+@receiver(restore_done, sender=Document,
+          dispatch_uid='api.document.restore_done.publish')
+def on_restore_done(sender, instance, **kwargs):
+    """
+    A signal handler to publish the document to the document API after it
+    has been restored.
+    """
+    publish.delay([instance.pk])
```

Comment:
It wouldn't hurt, but since the `restore_document` view isn't wrapped in `transaction.atomic`, it wouldn't make any difference.

Reply:
It should be :)
They should all be :)

Reply:
Actually, from what I can read if you do:

```python
@transaction.atomic
def view(request):
    transaction.on_commit(only_if_successful)
    return response
```
It *will* call `only_if_successful` only if the transaction is successful. But if you change it to this:

```python
# @transaction.atomic   COMMENTED OUT
def view(request):
    transaction.on_commit(only_if_successful)
    return response
```

then, it will act exactly the same as this:
```python
def view(request):
    only_if_successful()
    return response
```

I tested it with a Django 2.2 I had inside a side-project. One of my sample views looked like this:

```python
def log_transaction(line):
    with open('transactions.log', 'a') as f:
        f.write(line)
        f.write('\n')
    print("LOG TRANSACTION", time.time())

def view3(request):
    transaction.on_commit(lambda: log_transaction('view3'))
    sleep(2.1)
    print("In View 3")
    print(time.time())
    raise http.Http404
```
The output on stdout on this became:
```
LOG TRANSACTION 1556300548.503019
In View 3
1556300550.608419
```
which confirms what I mentioned above. Using `transaction.on_commit`, when not inside an atomic context, will work as it's not there. But once the `@transaction.atomic` is put in place, you get all sorts of juicy benefits such as NOT doing a S3 PUT if there's any Python exception in any of the other signals.

Reply:
But here, my suggestion does add risk.